lib/ocpp/v201/evse.cpp

```diff
@@ -135,6 +142,13 @@ void Evse::open_transaction(const std::string& transaction_id, const int32_t con
                 for (auto& item : meter_value.sampledValue) {
                     item.context = ReadingContextEnum::Sample_Clock;
                 }
+                // Get the aligned timestamp
+                auto old_time = meter_value.timestamp;
+                auto interval_value = std::chrono::seconds(
```

Comment:
You could also capture `aligned_data_tx_ended_interval` and then use that as the interval. Same for the other lambda